lib/src/ssh_signing.rs

```diff
@@ -81,8 +81,8 @@ fn run_command(command: &mut Command, stdin: &[u8]) -> SshResult<Vec<u8>> {
 fn ensure_key_as_file(key: &str) -> SshResult<Either<PathBuf, tempfile::TempPath>> {
     let is_inlined_ssh_key = key.starts_with("ssh-");
     if !is_inlined_ssh_key {
-        let key_path = Path::new(key);
-        return Ok(either::Left(key_path.to_path_buf()));
+        let key_path = crate::file_util::expand_home_path(key);
```

Comment:
(No action needed.) Ideally, path expansion should be restricted to config value, but that seems not doable because `sign_settings.key` may be interpreted differently by sign backends.